src/wix/WixToolset.Converters/WixConverter.cs

```diff
@@ -924,23 +923,28 @@ private void ConvertCustomTableElementToBundle(XElement element)
 
         private void ConvertControlElement(XElement element)
         {
-            var remove = new List<XElement>();
+            var xConditions = element.Elements(ConditionElementName).ToList();
 
-            foreach (var xCondition in element.Elements(ConditionElementName))
+            foreach (var xCondition in xConditions)
             {
                 var action = UppercaseFirstChar(xCondition.Attribute("Action")?.Value);
                 if (!String.IsNullOrEmpty(action) &&
-                    TryGetInnerText(xCondition, out var text) &&
+                    TryGetInnerText(xCondition, out var text, out var comments) &&
                     this.OnInformation(ConverterTestType.InnerTextDeprecated, element, "Using {0} element text is deprecated. Use the '{1}Condition' attribute instead.", xCondition.Name.LocalName, action))
                 {
                     element.Add(new XAttribute(action + "Condition", text));
-                    remove.Add(xCondition);
+                    // Comments are intentionally omitted here because there
```

Comment:
Could we instead leave the comments where they were rather than try to move them? At least then, no data is lost.